### PR TITLE
binned x axis, error bars and categorical y axis for line plot

### PR DESCRIPTION
**Pull request by d-callan**
mostly updating the schema, bit of refactoring w histo

### CODE REVIEW

**Comment by d-callan** [comment]
there may be more work on this yet. i havent decided how best to incorporate the data well want in the tooltips yet.

File: schema/url/visualizations.raml [hunk 1]
```diff
@@ -18,3 +18,14 @@ types:
     type: VariableSpec
     properties:
       value: string
+
+  # think we should assume type is binWidth if not provided?
+  # for line plot do we want to only allow binWidth? if so, we'll need to split this..
```
Comment (asizemore) [inline schema/url/visualizations.raml:23]: i think numBins makes sense for line plot. I have lots of points and i want 10. Seems like a reasonable desire.

File: src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java [hunk 2]
```diff
@@ -52,7 +54,7 @@ public ConstraintSpec getConstraintSpec() {
           .types(APIVariableType.NUMBER, APIVariableType.DATE, APIVariableType.INTEGER)
           .description("Variable must be a number or date.")
         .element("xAxisVariable")
-          .types(APIVariableType.NUMBER, APIVariableType.DATE, APIVariableType.INTEGER)
+          .shapes(APIVariableDataShape.ORDINAL, APIVariableDataShape.CONTINUOUS)
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java:57]: what happens with binning if a non-date, non-numeric ordinal var is used on the x axis? Like a var describing how sick you are with "not sick", "sick", "really sick"?

**Comment by d-callan** [comment]
also think i forgot binSpec in the response schema..

**Review by asizemore — COMMENTED** [review]
Looks good so far!

File: src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java [hunk 3]
```diff
@@ -321,6 +321,24 @@ protected String getViewportAsRString(XAxisViewport viewport, String xVarType) {
...
+  protected String listToRVector(List<String> values) {
+    boolean first = true;
+    String vector = "c(";
+
+    for (String value : values) {
+      if (first) {
+        vector = vector + singleQuote(value);
+        first = false;
+      } else {
+        vector = vector + singleQuote(value) + ",";
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java:334]: should be `vector + "," + singleQuote(value)` ? I think right now we'd get "('a''b','c',)"

File: src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java [hunk 4]
```diff
@@ -95,17 +97,40 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
...
+        } else {
+          binWidth = binSpec.getValue() == null ? "NULL" : "'" + binSpec.getValue().toString() + " " + binSpec.getUnits().toString().toLowerCase() + "'";
+        }
+        connection.voidEval("binWidth <- " + binWidth);
+      String cmd = "plot.data::lineplot(" + DEFAULT_SINGLE_STREAM_NAME + 
+                                        ", map, binWidth, " + 
+                                        singleQuote(valueSpec) + 
+                                        ", " + errorBars + 
+                                        ", viewport," + 
+                                        ", " + numeratorValues +
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java:131]: just checking - right now after `viewport` comes `evilMode` and `collectionVariablePlotRef` (looking at the plot.data PR). Are `numeratorValues` and `denominatorValues` for a later version?

**Review by asizemore — COMMENTED** [review]
Looking good! A few questions/notes here and there

File: src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java (hunk 2 above)
Comment (d-callan) [reply src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java:57]: i imagine wed ignore the bin width specified and issue a warning.. doesnt make sense to try to bin that, at least in this context. thatd be a derived variable. but i think that might be plot.data business, so ill make a note to add some unit tests etc there for the case.

**Review by asizemore — APPROVED** [review]
no comments outside of the valueSpec! Yay line plot!

**Pull request merged**